scripts/globals/combat/magic_hit_rate.lua

```diff
@@ -256,30 +263,40 @@ xi.combat.magicHitRate.calculateMagicHitRate = function(magicAcc, magicEva)
 end
 
 xi.combat.magicHitRate.calculateResistRate = function(actor, target, skillType, spellElement, magicHitRate)
-    local resistRate = 0
-    local resistRank = 0
+    local targetResistRate = 0 -- The variable we return.
+    local targetResistRank = 0
 
-    -- Magic Shield exception.
+    ----------------------------------------
+    -- Handle "Magic Shield" status effect.
+    ----------------------------------------
     if target:hasStatusEffect(xi.effect.MAGIC_SHIELD, 0) then
-        return resistRate
+        return targetResistRate
     end
 
-    -- Fetch resistance rank modifier.
+    ----------------------------------------
+    -- Handle target resistance rank.
+    ----------------------------------------
     if spellElement ~= xi.magic.ele.NONE then
-        resistRank = target:getMod(elementTable[spellElement][4])
+        targetResistRank = target:getMod(elementTable[spellElement][4])
     end
 
-    -- Resistance Ranks "boons".
-    if resistRank > 10 then -- Resistance rank 11 is technically the max, but we check for higher JUST IN CASE something altered it.
-        -- TODO: Inmunobreak logic probably goes here
+    -- Skillchains lowers target resistance rank by 1.
+    local _, skillchainCount = FormMagicBurst(spellElement, target)
```

Comment:
can you add a TODO around here regarding that Rayke needs to be implemented around here? (it lowers resistance rank)

Reply:
I can, however, wouldn't rayke apply a change to the base modifier itself, meaning the mod fetch would already include it?

Reply:
If that seems more appropriate, sure. I don't really mind where it's done, just a TODO for a reminder